compiler/analyzer.ml

```diff
@@ -137,6 +138,12 @@ let list_error list_type elem_type =
     (str_of_type elem_type) (str_of_type list_type) in
   raise (Semantic_Error message)
 
+let recursive_type_mismatch_error f_id expected_type typ = 
```

Comment:
If you pull from master-refactor I believe this error's already implemented as `fcall_argtype_error`. great minds think alike!